ffi-ctests/ctests/tests.c

```diff
@@ -0,0 +1,270 @@
+
+	wirefilter_free_scheme(scheme);
+	return true;
+}
+
+bool test_wirefilter_ffi_05() {
+	wirefilter_Scheme *scheme = wirefilter_create_scheme();
+	if(!scheme) {
+		return false;
+	}
+
+	wirefilter_ExternallyAllocatedStr field, filter_str;
+
+	WIREFILTER_STRING(field, "http.host");
+	wirefilter_add_type_field_to_scheme(scheme, field, wirefilter_Bytes);
+	WIREFILTER_STRING(field, "ip.addr");
+	wirefilter_add_type_field_to_scheme(scheme, field, wirefilter_Ip);
+	WIREFILTER_STRING(field, "ssl");
+	wirefilter_add_type_field_to_scheme(scheme, field, wirefilter_Bool);
+	WIREFILTER_STRING(field, "tcp.port");
+	wirefilter_add_type_field_to_scheme(scheme, field, wirefilter_Int);
+
+	WIREFILTER_STRING(filter_str, "tcp.port == 80");
+	wirefilter_ParsingResult result = wirefilter_parse_filter(scheme, filter_str);
+
+	if(!result.success) {
+		return false;
+	}
+
+	if(!result.ok.ast) {
+		return false;
+	}
+
+	wirefilter_Filter *filter = wirefilter_compile_filter(result.ok.ast);
+	if(!filter) {
+		return false;
+	}
+
+	wirefilter_free_compiled_filter(filter);
+
+	wirefilter_free_scheme(scheme);
+	return true;
+}
+
+bool test_wirefilter_ffi_06() {
+	wirefilter_Scheme *scheme = wirefilter_create_scheme();
+	if(!scheme) {
+		return false;
+	}
+
+	wirefilter_ExecutionContext *exec_ctx = wirefilter_create_execution_context(scheme);
+	if(!exec_ctx) {
+		return false;
+	}
+
+	wirefilter_free_execution_context(exec_ctx);
+
+	wirefilter_free_scheme(scheme);
+	return true;
+}
+
+bool test_wirefilter_ffi_07() {
+	wirefilter_Scheme *scheme = wirefilter_create_scheme();
+	if(!scheme) {
+		return false;
+	}
+
+	wirefilter_ExternallyAllocatedStr field;
+
+	WIREFILTER_STRING(field, "http.host");
+	wirefilter_add_type_field_to_scheme(scheme, field, wirefilter_Bytes);
+	WIREFILTER_STRING(field, "ip.addr");
+	wirefilter_add_type_field_to_scheme(scheme, field, wirefilter_Ip);
+	WIREFILTER_STRING(field, "ssl");
+	wirefilter_add_type_field_to_scheme(scheme, field, wirefilter_Bool);
+	WIREFILTER_STRING(field, "tcp.port");
+	wirefilter_add_type_field_to_scheme(scheme, field, wirefilter_Int);
+
+	wirefilter_ExecutionContext *exec_ctx = wirefilter_create_execution_context(scheme);
+	if(!exec_ctx) {
+		return false;
+	}
+
+	WIREFILTER_STRING(field, "http.host");
+	wirefilter_ExternallyAllocatedByteArr http_host;
+	WIREFILTER_STRING(http_host, "www.cloudflare.com")
+	wirefilter_add_bytes_value_to_execution_context(exec_ctx, field, http_host);
+
+	WIREFILTER_STRING(field, "ip.addr");
+	uint8_t ip_addr[4] = {192, 168, 0, 1};
+	wirefilter_add_ipv4_value_to_execution_context(exec_ctx, field, ip_addr);
+
+	WIREFILTER_STRING(field, "ssl");
+	wirefilter_add_bool_value_to_execution_context(exec_ctx, field, false);
+
+	WIREFILTER_STRING(field, "tcp.port");
+	wirefilter_add_int_value_to_execution_context(exec_ctx, field, 80);
+
+	wirefilter_free_execution_context(exec_ctx);
+
+	wirefilter_free_scheme(scheme);
+	return true;
+}
+
+bool test_wirefilter_ffi_08() {
```

Comment:
This naming is confusing. What tests 01-08 actually testing?

Reply:
Should i just rename each function with a more descriptive name or just keep the last one that pretty much test everything?